invisible_cities/cities/detsim_get_psf.py

```diff
@@ -0,0 +1,66 @@
+import numpy  as np
+import tables as tb
+import pandas as pd
+
+from typing import Tuple, Callable
+
+from invisible_cities.reco.corrections_new import read_maps
+
+##################################
+############# PSF ################
+##################################
+def _psf(dx, dy, dz, factor = 1.):
+    """ generic analytic PSF function
+    """
+    return factor * np.abs(dz) / (2 * np.pi) / (dx**2 + dy**2 + dz**2)**1.5
+
+
+def create_xy_function(array : np.array,
+                       xaxis : Tuple[float],
+                       yaxis : Tuple[float])->Callable:
+    xmin, xmax, dx = xaxis
+    ymin, ymax, dy = yaxis
+    def function(x, y, z=0):
+        x = np.clip(x, xmin, xmax-0.1*dx)
+        y = np.clip(y, ymin, ymax-0.1*dy)
+        ix = ((x-xmin)//dx).astype(int)
+        iy = ((y-ymin)//dy).astype(int)
+        return array[ix, iy]
```

Comment:
I'd prefer it had a more explicative name, function seems a bit generic as a name.

Also, you don't use z. Did you forget to push part of the code or is z not used? It shouldn't be an argument of the function if it has no use.

Also the type hint for a numpy array is np.ndarray. I'd give it a descriptive name other than just array which is confusing.

Reply:
Now, I changed this `create_xy_function` by `create_xyx_function`.  The fact of adding the `z` variable is because when we compute `pes = photons * psf(dxs, dys, dzs)` (PR #697 in `pes_at_sensors ` function), this `psf` is in general a x, y, z function. 

For krmap and sipmpsf, z dimension is not used, but it is added to do not include something duplicate like:
```
if S2:
    pes = photons * psf(dxs, dys, dzs)
elif S1:
    pes = photons * psf(dxs, dys)
```
Besides, we can add light tables, which are x, y, z functions.